plugin-server/tests/e2e.test.ts

```diff
@@ -118,20 +122,26 @@ describe('E2E', () => {
             source__index_ts: indexJs,
         })
 
+        pluginsServers = await pluginServer()
+    })
+
+    beforeEach(async () => {
+        testConsole.reset()
+        teamId = await createTeam(postgres, organizationId)
         pluginConfig = await createPluginConfig(postgres, { team_id: teamId, plugin_id: plugin.id })
         // Make sure the plugin server reloads the newly created plugin config.
         // TODO: avoid reaching into the pluginsServer internals and rather use
         // the pubsub mechanism to trigger this.
-        await pluginsServer.piscina?.broadcastTask({ task: 'reloadPlugins' })
+        await Promise.all(pluginsServers.map((instance) => instance.piscina.broadcastTask({ task: 'reloadPlugins' })))
     })
 
     afterAll(async () => {
-        await pluginsServer.stop()
+        await Promise.all(pluginsServers.map((instance) => instance.stop()))
         await producer.disconnect()
         await postgres.end()
     })
 
-    describe('ClickHouse ingestion', () => {
+    describe(`ClickHouse ingestion (${pluginServer.name})`, () => {
```

Comment:
This makes sense now I guess but breaks down once we split things down further, no?

Reply:
We would define further instantiation functions, or just update the existing one.